src/docfx/Errors.cs

```diff
@@ -73,8 +73,8 @@ public static Error ConfigNotFound(string docsetPath)
         ///   - a.md references b.json's property with xref syntax, and b.json includes a.md reversely
         /// </summary>
         /// Behavior: ✔️ Message: ✔️
-        public static Error CircularReference<T>(IEnumerable<T> dependencyChain)
-            => new Error(ErrorLevel.Error, "circular-reference", $"Build has identified file(s) referencing each other: {string.Join(" --> ", dependencyChain.Select(file => $"'{file}'"))}");
+        public static Error CircularReference<T>(IEnumerable<T> dependencyChain, Document declaringFile)
+            => new Error(ErrorLevel.Error, "circular-reference", $"Build has identified file(s) referencing each other: {string.Join(" --> ", dependencyChain.Select(file => $"'{file}'"))}", file: declaringFile.FilePath);
```

Comment:
```
{string.Join(" --> ", dependencyChain.Select(file => $"'{file}'")) + $"--> {declaringFile}"}"
```

Reply:
This would not work, the dependencyChain should be handled by the caller. We have usage in TOC and Xref, they are constructed in different ways

Reply:
we should be using the same way to detect circular reference using a thread local stack

Reply:
updated